litespi/spi_nor_config.py

```diff
@@ -0,0 +1,29 @@
+import enum
+
+
+class SpiNorConfig(enum.Flag):
+    """SPI NOR Config
+
+    The ``SpiNorConfig`` class provides enums used to describe SPI NOR
+    configuration and chip capabilites.
```

Comment:
Isn't this covered by the `SpiNorFeatures` class?

Reply:
Partially yes, there are some missing features. I will add them there and swap SpiNorConfig to SpiNorFeatures

Reply:
I think I had most of the important features at https://github.com/mithro/litex/blob/d7184b97494012a8c21ee93df231b758a61e9dec/litex/soc/cores/spi/modules.py#L141-L193 ?

Reply:
That's correct, but I would like to add to `SpiNorFeatures` class such flags as:
- Dual mode support
- Octal mode support
- Fast read mode (24bit)

In `SpiNorFeatures` there is a flag for Quad Mode (`FEATURE_QPI`) so adding `FEATURE_DUAL` and `FEATURE_OCTAL` would be complementary.
Additionally there is `FEATURE_4BA_FAST_READ` but it lacks of 24-bit fast read.

Reply:
Dual mode is not clear. There are multiple "dual" modes;
 -     `READ_1_1_2        = _Op(0x3b, "Read data bytes (Dual Output SPI)")`
 -     `READ_1_2_2        = _Op(0xbb, "Read data bytes (Dual I/O SPI)")`
 -     `READ_1_2_2_DTR    = _Op(0xbd, "")`

Reply:
Will be reworked in the next PR